### ISSUE TITLE
The ISIC training dice is stuck at 0.38

### ISSUE THREAD

**Issue by xyimaging** [Digital-Dermatology/t-loss#3]
Hi authors, thanks for your good work and congrats on the best paper award!

I'm trying to reimplement the training on ISIC with alpha = 0.7 and beta = 0.7 following your instruction, at the first several epochs the training seems to be ok and dice kept increasing, however, after the epoch 12 the dice fell off a cliff and was stuck at 0.38. Could you please give any insight about this? Thanks!
![image](https://github.com/Digital-Dermatology/t-loss/assets/56007678/be0139f1-3b10-4341-85ea-c5349ae0dc20)
![image](https://github.com/Digital-Dermatology/t-loss/assets/56007678/26879ab0-722d-4156-adeb-2ccc03bf3228)
![image](https://github.com/Digital-Dermatology/t-loss/assets/56007678/7f4a2ca7-7ba2-4f40-91cd-f7bf4e5946af)

**Comment by alvarogonjim** [comment]
Hi @xyimaging, thank you for your interest in our work. To ensure the reproducibility of the results, we recommend using our own implementation.

Regarding your question, a possible explanation is that in your implementation, $\nu$ reaches a plateau, but the model continues to push it to lower values. This suggests that the model might be overfitting to the noisy labels (resulting in a drop of the Dice score to 0.38). To address this issue, you could record the values of $\nu$ at the iteration level, as we do, and adapt the learning rate based on the behavior of $\nu. Please refer to: https://github.com/Digital-Dermatology/t-loss/blob/f132e8767c2191463bbcc2c15310cd8df1b7ab30/train.py#L312-L323

**Comment by xyimaging** [comment]
Thanks for your quick reply! I'll have a try.
